docs/kb/rules/match-commments.md

```diff
@@ -0,0 +1,78 @@
+---
+description: Semgrep's generic pattern matching mode can match comments in code files.
+tags:
+  - Rules
+  - Semgrep Code
+---
+
+# Match comments with Semgrep
+
+When Semgrep rules target a language or languages, they do not match comments in the targeted code files. Comments are not part of the semantic and syntactic structure of the document, so they are ignored.
```

Comment:
```suggestion
When Semgrep rules target languages, they do not match comments in the targeted code files. Comments are not part of the semantic and syntactic structure of the document, so they are ignored.
```

Reply:
good call that this is awkward - I think I might go with "target specific languages" to make clear the opposition to `generic`

Reply:
maybe I can just suggest things to myself, instead of doing the proper git thing... 😂